### PR TITLE
Url signing for the image service

### PR DESCRIPTION
**Pull request by gklopper**

### CODE REVIEW

File: common/app/implicits/Requests.scala [hunk 1]
```diff
@@ -32,7 +32,16 @@ trait Requests {
...
+
+    private val imgixTestSections = Seq(
+      "/uk/money",
+      "/au/money",
+      "/us/money",
+      "/money"
+    )
+
+    lazy val isInImgixTest: Boolean = Switches.ImgixSwitch.isSwitchedOn &&
+      (Configuration.environment.isNonProd || imgixTestSections.exists(r.path.startsWith))
```
Comment (johnduffell) [inline common/app/implicits/Requests.scala]: surely you can just use the switchboard to make it on everywhere except prod ?

Comment (gklopper) [inline common/app/implicits/Requests.scala]: Slightly more subtle than that.

The switch is binary on/ off - this means it is limited to whitelisted sections in PROD but on in all sections elsewhere.

**Comment by robertberry** [comment]
:+1:

**Comment by robertberry** [comment]
Shame there's no decent URI manipulation library in Scala yet. See lots of code basically doing String manipulation where it would be nice to have methods for manipulating a URI ...

**Comment by johnduffell** [comment]
:+1: @robertberry I thought the same about if url contains a '?' then do this else that.  That's the source of all security holes, hacky string manipulation is (well, not to forget human factors etc)

File: common/app/implicits/Requests.scala (hunk 1 above)
Comment (johnduffell) [inline common/app/implicits/Requests.scala]: ok fair enough, that condition still made my eyes hurt! :confused:

**Comment by robertberry** [comment]
@johnduffell Yup. Unfortunately although Java has a `URI` class, it doesn't provide any nice way of copying with some bits changed.

**Comment by robertberry** [comment]
@johnduffell And Java's URL class is completely crazy. If you do two completely different URLs that resolve to the same IP address for the host and have the same path, it considers them identical. (Just think about how dumb that is!)

**Comment by johnduffell** [comment]
@robertberry well it considers them .equal but not == right?  and that's the joy of java's equals method, it might do anything

**Comment by robertberry** [comment]
To be fair you can override `equals` in Scala, too - it's just case classes give it a sensible default

**Pull request merged**